soc/nxp/mcx/mcxw/CMakeLists.txt

```
@@ -2,18 +2,34 @@
 #
 # SPDX-License-Identifier: Apache-2.0
 
-zephyr_sources(soc.c)
-zephyr_sources_ifdef(CONFIG_SOC_MCXW716C mcxw71_platform_init.S)
-zephyr_sources_ifdef(CONFIG_SOC_MCXW727C mcxw72_platform_init.S)
+if (CONFIG_SOC_MCXW716C OR CONFIG_SOC_MCXW727C)
+  zephyr_sources(mcxw7x/soc.c)
+  zephyr_sources_ifdef(CONFIG_SOC_MCXW716C mcxw7x/mcxw71_platform_init.S)
+  zephyr_sources_ifdef(CONFIG_SOC_MCXW727C mcxw7x/mcxw72_platform_init.S)
 
-zephyr_sources_ifdef(CONFIG_NXP_NBU
-  ../../common/nxp_nbu.c
-  )
+  zephyr_sources_ifdef(CONFIG_NXP_NBU
+    ../../common/nxp_nbu.c
+    )
 
-zephyr_include_directories(.)
+  zephyr_include_directories(./mcxw7x)
 
-set(SOC_LINKER_SCRIPT ${CMAKE_CURRENT_SOURCE_DIR}/linker.ld CACHE INTERNAL "")
+  set(SOC_LINKER_SCRIPT ${CMAKE_CURRENT_SOURCE_DIR}/mcxw7x/linker.ld CACHE INTERNAL "")
 
-if (CONFIG_BT OR CONFIG_IEEE802154)
-  zephyr_linker_sources(RAM_SECTIONS sections.ld)
+  if (CONFIG_BT OR CONFIG_IEEE802154)
```

```suggestion
  if(CONFIG_BT OR CONFIG_IEEE802154)
```
no spaces between if and bracket, fix below also